exchange/bidder.go

```diff
@@ -87,20 +88,27 @@ type pbsOrtbSeatBid struct {
 //
 // The name refers to the "Adapter" architecture pattern, and should not be confused with a Prebid "Adapter"
 // (which is being phased out and replaced by Bidder for OpenRTB auctions)
-func adaptBidder(bidder adapters.Bidder, client *http.Client, cfg *config.Configuration, me pbsmetrics.MetricsEngine) adaptedBidder {
+func adaptBidder(bidder adapters.Bidder, client *http.Client, cfg *config.Configuration, me pbsmetrics.MetricsEngine, name openrtb_ext.BidderName) adaptedBidder {
 	return &bidderAdapter{
 		Bidder:      bidder,
+		BidderName:  name,
 		Client:      client,
 		DebugConfig: cfg.Debug,
-		me:          me,
+		me:          bidderMetrics{engine: me, disabledMetrics: cfg.Metrics.Disabled},
 	}
 }
 
 type bidderAdapter struct {
 	Bidder      adapters.Bidder
+	BidderName  openrtb_ext.BidderName
 	Client      *http.Client
 	DebugConfig config.Debug
-	me          pbsmetrics.MetricsEngine
+	me          bidderMetrics
+}
+
+type bidderMetrics struct {
+	engine          pbsmetrics.MetricsEngine
+	disabledMetrics config.DisabledMetrics
```

Comment:
Do we need to have the `disabledMetrics` info at the bidder level? Right now the way it is configured, we can either enable connection metrics for all adapter or no adapters and so I am wondering if this is needed at the bidder level?

Reply:
It was suggested in one the meetings we held. In the future we'll probably want to collect metrics only for certain adapters.

Reply:
You mean consolidating the `DebugConfig` and the `disabledMetrics` fields? 

So go from this: (current version)
```
type bidderAdapter struct {
    Bidder      adapters.Bidder
    Client      *http.Client
    DebugConfig config.Debug
    me          pbsmetrics.MetricsEngine
}
```

To this: 
```
type bidderAdapter struct {
    Bidder      adapters.Bidder
    Client      *http.Client
    me          pbsmetrics.MetricsEngine
    config      bidderAdapterConfig
}
type bidderAdapterConfig struct {
    DebugConfig   config.Debug
    MetricsConfig config.DisabledMetrics
}
```

And not this: (proposed in PR)
```
type bidderAdapter struct {
    Bidder      adapters.Bidder
    BidderName  openrtb_ext.BidderName
    Client      *http.Client
    DebugConfig config.Debug
    me          bidderMetrics
}

type bidderMetrics struct {
    engine          pbsmetrics.MetricsEngine
    disabledMetrics config.DisabledMetrics
}
```

Did I get it right?

Reply:
Yup! That's right. Although, I think rather than having the entire `DisabledMetrics` struct in `bidderAdapterConfig`, it might be better to have just a boolean for connection metrics that indicates if connection metrics for this bidder is enabled or not (Right now, it'd be true for all bidders). Because we aren't using the `AccountAdapterDetails` value anywhere here and also if we add more metrics to the `DisabledMetrics` struct in the future, not each of them would make sense at the bidder level and so we might wanna keep only the ones we're interested in over here rather than copying the entire `DisabledMetrics` struct.

Reply:
Done